File: src/main/java/net/theevilreaper/vulpes/api/model/AttributeModel.java
package net.theevilreaper.vulpes.api.model;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.serde.annotation.Serdeable;

/**
 * The model class describes the data structure to represent an attribute from the game Minecraft.
 *
 * @author theEvilReaper
 * @since 1.0.0
 */
@Serdeable
@MappedEntity("attributes")
public record AttributeModel(
        @Id String id,
        String modelName,
        String name,
        double defaultValue,
        double maximumValue
) implements VulpesModel {
}

File: src/main/kotlin/net/theevilreaper/vulpes/api/model/AttributeModel.kt
package net.theevilreaper.vulpes.api.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

/**
 * The model class describes the data structure to represent an attribute from the game Minecraft.
 * @since 1.0.0
 * @author theEvilReaper
 */
@Document("attributes")
data class AttributeModel(
    @Id
    val id: String?,
    val modelName: String?,
    val name: String?,
    val defaultValue: Double?,
    val maximumValue: Double?
) : VulpesModel